ApollosProject/apollos-apps
# Interactions schema

## DESCRIPTION

Companion PR, to be merged after: https://github.com/ApollosProject/apollos-templates/pull/2

### What does this PR do, or why is it needed?

Adds a `mutation` to create an `interaction` against a generic NodeId. That NodeId can be any type of Entity, but it will work best if it's a Rock Entity (for now, anyway). 

### What design trade-offs/decisions were made?

I cleaned up a lot of older code to hopefully make the purpose of this code more clear. 

### How do I test this PR?
1. Boot this PR in apollos-templates master
2. Login with this mutation
```
mutation login {
  authenticate(identity:"vincent@differential.com", password:"password") {
    token
  }
}
```
(store your token inside the Authorization header)
3. Run this mutation in graphiql
```
mutation {
  interactWithNode(nodeId:"WeekendContentItem:2b32682bd0d96179e6f001060251c96f", action: COMPLETE){
    success
    node {
      id
    }
  }
}
```
4. You should see your mutation show up here:
https://apollosrock.newspring.cc/page/510?ComponentId=263

### What automated tests did you write?

Tests that cover new functionality, and edited older tests to better cover all functionality.

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
